packages/app-mobile/components/SideMenu.tsx

```diff
@@ -0,0 +1,73 @@
+const React = require('react');
+const { connect } = require('react-redux');
+const SideMenu_ = require('react-native-side-menu-updated').default;
+import { Dimensions, DrawerLayoutAndroid, Platform } from 'react-native';
+import { State } from '@joplin/lib/reducer';
+import { FunctionComponent, useEffect, useRef } from 'react';
+
+class SideMenuFallback extends SideMenu_ {
+	public onLayoutChange(e: any) {
+		const { width, height } = e.nativeEvent.layout;
+		const openMenuOffsetPercentage = this.props.openMenuOffset / Dimensions.get('window').width;
+		const openMenuOffset = width * openMenuOffsetPercentage;
+		const hiddenMenuOffset = width * this.state.hiddenMenuOffsetPercentage;
+		this.setState({ width, height, openMenuOffset, hiddenMenuOffset });
+	}
+}
+
+const SideMenuComponent: FunctionComponent<any> = ({
+	children,
+	...props
+}) => {
+	const androidLayoutRef = useRef<DrawerLayoutAndroid>();
+
+	useEffect(() => {
+		if (androidLayoutRef.current) {
+			if (props.isOpen) {
+				androidLayoutRef.current.openDrawer();
+			} else {
+				androidLayoutRef.current.closeDrawer();
+			}
+		}
+	}, [props.isOpen, androidLayoutRef]);
+
+	if (Platform.OS === 'android') {
+		return (
+			<DrawerLayoutAndroid
+				// Need to do this if the drawer position switches from left to right
+				// Otherwise, there's a bug
+				// To reproduce:
+				// - Open menu with no note open
+				// - Close menu
+				// - Open note
+				// - Open menu from the right
+				// - Menu flashes into open state
+				key={`drawer-${props.menuPosition}`}
+				ref={androidLayoutRef}
+				renderNavigationView={() => props.menu}
+				drawerPosition={props.menuPosition}
+				drawerWidth={props.openMenuOffset}
+				onDrawerOpen={() => (props.onDrawerStateChange && props.onDrawerStateChange(true))}
+				onDrawerClose={() => (props.onDrawerStateChange && props.onDrawerStateChange(false))}
+				{...props}
+			>
+				{children}
+			</DrawerLayoutAndroid>
+		);
+	}
+
+	return (
+		// @ts-ignore
```

Comment:
We don't want `@ts-ignore` either. What was the warning here?

Reply:
@laurent22 Well it's giving me this:

![image](https://user-images.githubusercontent.com/1340627/227195695-689c6074-a3e8-4711-94ab-37c2801fbd7a.png)

I think it's down to the fact that the export from the react-native-side-menu library is not a TS library. I'm not sure how to get around this.

Reply:
Hmm, yes it's old and rather terrible code because this SideMenuFallback extends another component, which we definitely shouldn't do. Let's leave the ts-ignore then, but please can you add a comment to explain why it's there?

Reply:
ok added in 312bbb8f0